pytentiostat/tester.py

```diff
@@ -87,7 +87,8 @@ def read_write(
     times_diff_list = [x - starting_time for x in times_list]
     times_diff_list.append(0)
     t = 1
-    
+    times, voltages, currents = [], [], []

```

Comment:
Before I was calling this globally to work between the two main functions in this file. Does this still allow cv to work with the real time plotting? Also would check exported data.

Reply:
Yes, you're right. If I overwrite the global variables, then the saved data is blank. Let me fix that.

Reply:
Fixed. I had to move the global arrays into the experiment function and then pass them to read_write and return them back to experiment.

Reply:
👍 

Maybe good test for this in future can be trying to run def experiment(): and mock reading a single non-zero entry for each list in read_write. Expected output should match the single-nonzero entry and end lists should have lengths of how many times read_write was called.

Reply:
So I learned yesterday with  @weiziyuan while working on the GUI that even if you don't call a list globally, when you operate on a list inside another function, the list stays changed. So like we pass the lists to experiment, and experiment uses them as inputs for read/write. Read write appends to them and even without returning them they stay modified in the context of the experiment function. So they can be returned to main from experiment even without having them be returned from read_write to experiment. 

So it's more explicit here which might make more easy to follow, but I guess could be made more compact by utilizing that property.